src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,79 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<Integer> supplyAmounts = new ArrayList<>();
+        List<Integer> buyAmounts = new ArrayList<>();
+
+        readAndParseData(fromFileName, supplyAmounts, buyAmounts);
+
+        int[] statistics = calculateStatistics(supplyAmounts, buyAmounts);
+
+        writeResults(toFileName, statistics);
```

Comment:
Local variables `supplyAmounts` and `buyAmounts` are being used to carry data through the workflow. Consider using local variables inside the methods where they are processed instead of passing them through method parameters if they are not intended to be shared states of the object.